src/RingBufferLib.sol

```diff
@@ -76,4 +76,16 @@ library RingBufferLib {
     {
         return wrap(_index + 1, _cardinality);
     }
+
+    /// @notice Computes the ring buffer index that preceeds the given one, wrapped by cardinality
+    /// @param _index The index to increment
+    /// @param _cardinality The number of elements in the Ring Buffer
+    /// @return The prev index relative to the given index.  Will wrap around to the end if the prev index == 0
+    function prevIndex(uint256 _index, uint256 _cardinality)
+    internal
+    pure
+    returns (uint256) 
+    {
+        return wrap(_index == 0 ? _cardinality - 1 : _index - 1, _cardinality);
```

Comment:
Don't think you need to wrap here, as the index - 1 is always less than cardinality